src/lsstseries/ensemble.py

```diff
@@ -284,6 +284,66 @@ def prune(self, threshold=50, col_name=None):
 
         return self
 
+    def bin_sources(self, time_window=1.0, additional_cols=None, use_map=True, **kwargs):
+        """Bin sources on within a given time range to improve the estimates.
+
+        Notes
+        -----
+        * This should only be used for slowly varying sources where we can
+        treat the source as constant within `time_window`.
+
+        * As a default the function only aggregates and keeps the id, band,
+        time, flux, and flux error columns. Additional columns can be preserved
+        by providing the mapping of column name to aggregation function with the
+        `additional_cols` parameter.
+
+        Parameters
+        ----------
+        time_window : `float` (optional)
+            The time range (in days) over which to consider observations in the same bin.
+        additional_cols : `dict` (optional)
+            A dictionary mapping column name to aggregation method.
+            Example: {"my_value_1": "mean", "my_value_2": "max"}
+        use_map : `boolean` (optional)
+            Determines whether `dask.dataframe.DataFrame.map_partitions` is
+            used (True). Using map_partitions is generally more efficient, but
+            requires the data from each lightcurve is housed in a single
+            partition. If False, a groupby will be performed instead.
+        """
+        # Bin the time and add it as a column.
+        tmp_time_col = "tmp_time_for_aggregation"
+        if tmp_time_col in self._source.columns:
+            raise KeyError(f"Column '{tmp_time_col}' already exists in source table.")
+        self._source[tmp_time_col] = self._source[self._time_col].apply(
+            lambda x: np.floor(x / time_window) * time_window, meta=pd.Series(dtype=float)
+        )
+
+        # Set up the aggregation functions for the time and flux columns.
+        aggr_funs = {self._time_col: "mean", self._flux_col: "mean"}
+
+        # If the source table has errors then add an aggregation function for it.
+        if self._err_col in self._source.columns:
+            aggr_funs[self._err_col] = dd.Aggregation(
+                name="err_agg",
+                chunk=lambda x: (x.count(), x.apply(lambda s: np.sum(np.power(s, 2)))),
+                agg=lambda c, s: (c.sum(), s.sum()),
+                finalize=lambda c, s: np.sqrt(s) / c,
+            )
+
+        # Add any additional aggregation functions
+        if additional_cols is not None:
```

Comment:
What happens if one of (self._flux_col, self._time_col, self._err_col) are provided in additional_cols? It looks like it might clobber the assigned aggregations, which is kind of what I was asking about above, but this may not be intended?

Reply:
This was not originally my intention, but I think we should leave it in given your question above. I changed the variable name to reflect this and expanded the docstring.

Reply:
I see you answered this above, but this may not be intended, if technically the error calculation should be dependent on the flux calculation. I could see someone using this for good, for example if they want custom aggregations for their errors. But also bad, as someone may update the flux aggregation and not realize the errors won't adapt. Maybe we should just throw a warning if the core aggregations would be overwritten?